csrc/multidevice/lower_communication.cpp

```diff
@@ -155,11 +157,16 @@ void lowerToBroadcast(
         " vs ",
         receiver_mesh.size());
     for (auto i : c10::irange(sender_mesh.size())) {
-      lowerToBroadcast(
-          my_device_index,
-          sender_mesh.at(i),
-          DeviceMesh({receiver_mesh.at(i)}),
-          comms);
+      const DeviceIdxType sender = sender_mesh.at(i);
```

Comment:
I think this change is beneficial but I'm open to ideas. 

Consider a collective send/recv from `mesh{0,1,2,3}` to `mesh{4,5,6,7}` (for pipeline parallelism) or from `mesh{0,1,2,3}` to `mesh{1,2,3,0}` (for p2p ring), that's decomposed into four ncclSend/ncclRecv pairs. 

Currently, send/recv pairs differ in `mesh` and therefore `team`. With this PR, they will differ in `team` but have the same input and output meshes, allowing them to reuse the same input and output TensorViews. 

(These examples also make me think about letting SendRecv take a vector of send/recv pairs. But that's out of the scope of this PR.)

Reply:
> (These examples also make me think about letting SendRecv take a vector of send/recv pairs. This way, one SendRev does all and therefore shortens the host IR. But that's out of the scope of this PR.)

Also out of the scope of the PR :) 
Since these programs are SPMD like why not just use an expression to calculate one send/recv pairs instead of expanding out everything? For the pipeline parallelism example 0 sends to 4, 1 to 5 etc. Once a device has been determined to be to be in a pipeline stage then it's guaranteed to be either a sender or receiver for one of many parallel communications.